src/mbgl/renderer/painter_debug.cpp

```diff
@@ -5,162 +5,59 @@
         }
-        if (frame.debugOptions & MapDebugOptions::TileBorders) {
-            renderDebugFrame(tile.matrix);
-        }
-    }
-}
-
-void Painter::renderDebugText(Tile& tile, const mat4 &matrix) {
-    MBGL_DEBUG_GROUP("debug text");
 
-    context.depthTest = false;
+        const auto& vertexBuffer = tile.debugBucket->vertexBuffer;
 
-    if (!tile.debugBucket || tile.debugBucket->renderable != tile.isRenderable() ||
-        tile.debugBucket->complete != tile.isComplete() ||
-        !(tile.debugBucket->modified == tile.modified) ||
-        !(tile.debugBucket->expires == tile.expires) ||
-        tile.debugBucket->debugMode != frame.debugOptions) {
-        tile.debugBucket = std::make_unique<DebugBucket>(
-            tile.id, tile.isRenderable(), tile.isComplete(), tile.modified,
-            tile.expires, frame.debugOptions, context);
+        draw(Color::white(), gl::Unindexed<gl::Lines>(vertexBuffer, 4.0f * frame.pixelRatio));
+        draw(Color::black(), gl::Unindexed<gl::Points>(vertexBuffer, 2.0f));
+        draw(Color::black(), gl::Unindexed<gl::Lines>(vertexBuffer, 2.0f * frame.pixelRatio));
     }
 
-    auto& plainShader = shaders->fill;
-    context.program = plainShader.getID();
-    plainShader.u_matrix = matrix;
-    plainShader.u_opacity = 1.0f;
-
-    // Draw white outline
-    plainShader.u_color = Color::white();
-    context.lineWidth = 4.0f * frame.pixelRatio;
-    tile.debugBucket->drawLines(plainShader, context);
-
-#if not MBGL_USE_GLES2
-    // Draw line "end caps"
-    MBGL_CHECK_ERROR(glPointSize(2));
-    tile.debugBucket->drawPoints(plainShader, context);
-#endif // MBGL_USE_GLES2
-
-    // Draw black text.
-    plainShader.u_color = Color::black();
-    context.lineWidth = 2.0f * frame.pixelRatio;
-    tile.debugBucket->drawLines(plainShader, context);
-
-    context.depthFunc = gl::DepthTestFunction::LessEqual;
-    context.depthTest = true;
-}
-
-void Painter::renderDebugFrame(const mat4 &matrix) {
-    MBGL_DEBUG_GROUP("debug frame");
-
-    // Disable depth test and don't count this towards the depth buffer,
-    // but *don't* disable stencil test, as we want to clip the red tile border
-    // to the tile viewport.
-    context.depthTest = false;
-    context.stencilOp = { gl::StencilTestOperation::Keep, gl::StencilTestOperation::Keep,
-                          gl::StencilTestOperation::Replace };
-    context.stencilTest = true;
-
-    auto& plainShader = shaders->fill;
-    context.program = plainShader.getID();
-    plainShader.u_matrix = matrix;
-    plainShader.u_opacity = 1.0f;
-
-    // draw tile outline
-    tileBorderArray.bind(plainShader, tileLineStripVertexBuffer, BUFFER_OFFSET_0, context);
-    plainShader.u_color = { 1.0f, 0.0f, 0.0f, 1.0f };
-    context.lineWidth = 4.0f * frame.pixelRatio;
-    MBGL_CHECK_ERROR(glDrawArrays(GL_LINE_STRIP, 0, static_cast<GLsizei>(tileLineStripVertexBuffer.vertexCount)));
-}
-
-#ifndef NDEBUG
```

Comment:
I removed `Painter::renderClipMasks` and `Painter::renderDepthBuffer` in an early step to avoid dealing with them at the time, and need to go back and convert them now.